### DISCUSSION TITLE
widget compatibility with WP 5.8

### DISCUSSION THREAD

**Discussion by bugalug** [code4recovery/12-step-meeting-list#523] [Q&A] [answered]
There appears to be a compatibility problem with widgets since WP 5.7 - I am seeing the issue reported by many people in other plugin forums. I didn't notice it until now, but the "Upcoming Meetings" function no longer works.

It will throw errors when I look at widgets. Error example: "The "tsml_widget_upcoming" block was affected by errors and may not function properly. Check the developer tools for more details."

Other widgets sometimes continue working on the sites despite errors, but in the case of TSML they are not. In my case for TSML the broken widget seems to be "Upcoming Meetings."

Sometimes the simpler ones (like "custom HTML" widgets) can be fixed by just removing and re-creating them as blocks.

Unfortunately, that is not working for this problem. Re-creating it just brings back the errors. Any help is appreciated.

**Comment by joshreisner** [comment]
Hi @bugalug thanks for reporting the issue - I'm having trouble reproducing it. We're running the widget with WP `5.8.1` on our demo site at https://demo.code4recovery.org, looks ok to me. 

Our Appearance > Widgets page looks ok too.

<img width="1468" alt="Screen Shot 2021-09-11 at 7 24 09 AM" src="https://user-images.githubusercontent.com/1551689/132951132-0cf4baa7-caea-4746-a97d-285b9d3781cb.png">

Could you provide a screenshot of the issue or a URL?

**Reply by bugalug** [reply]
<img width="1161" alt="tsml widget problem" src="https://user-images.githubusercontent.com/90507760/132962040-107e6b44-7bb6-421e-897b-ed7abc348195.png">

**Reply by bugalug** [reply]
The theme is twenty-twenty, no child. Here are the plugins.

<img width="1103" alt="plugins" src="https://user-images.githubusercontent.com/90507760/132962111-0c266b85-878e-4fe6-aeb7-2accbc139a06.png">

**Reply by joshreisner** [reply]
Hm, so I still can't reproduce this issue locally. I'm using Twenty Twenty, I added the widget to the Location Detail Bottom widget area like you did, and I even imported your data to my demo site (on my personal machine, not the internet).

<img width="1437" alt="widgets" src="https://user-images.githubusercontent.com/1551689/132993194-27a41965-63d0-4f19-b5a6-c06a9340d175.png">

<img width="1437" alt="location-bottom" src="https://user-images.githubusercontent.com/1551689/132993195-23757a5c-4c49-4d05-bed4-972d460635e0.png">

When they say "check the developer tools" I'm assuming they mean the Javascript console. Could you [open the javascript console on your browser](https://balsamiq.com/support/faqs/browserconsole/) and see if there's any info in there?

**Comment by Grateful-Viv** [comment, marked as answer]
Try using the classic widget plugin. I found when upgrading to 5.8 that the new widget blocks broke most of my html widgets as it tried to convert them to these ugly blocks. It didn’t just affect tsml. Once broken you will have to redo it unless you can restore from backup.

**Reply by bugalug** [reply]
Yes, as I said, it affected Formidable Forms and even Custom HTML also, but once I removed and re-added those they are fine. I just thought that the developers of TSML would be working on a fix for their product, so that we would not need to treat it as a "legacy" plugin.

**Reply by Grateful-Viv** [reply]
The developers do not have to be working on a fix for their product and this widget block fiasco is more on the WordPress Team than any plugin author. Why WordPress thought this was a good idea to force feed people this sort of stuff (and without warning is terrible). I see the WordPress trend is going to this ugly Full Site Editing (FSE or Page Block Editing) concept, and it is going to make it more difficult hard to have any sort of hybrid theme. I just hope it takes a really long time before they truly destroy their legacy software.

But again, if you're upset that your custom HTML was suddenly broken, get mad and complain to WordPress. They gave no warning this was coming. You don't need the plugin to do what you want to do for your widgets - just use the new block editor. 

Personally, I do not do Gutenberg and am using Classic Editor to prevent it. There are other plugins besides the Classic one you could use that work for whatever options you want such as [Disable Gutenberg](https://wordpress.org/plugins/disable-gutenberg/).

But please do not blame the TSML developers. This one is not on them.

**Reply by joshreisner** [reply]
Thanks @Grateful-Viv I appreciate the support! I think our official position is we do want to support WP out of the box, no matter what questionable decisions they may make :)

I would be curious to hear if the classic widgets plugin does fix the issue.

**Reply by Grateful-Viv** [reply]
What WP 5.8 did was break any custom HTML used in widgets. It tried to turn them all into blocks and did a terrible job of it. Besides the meeting top widget, I have four, rather complicated footer widgets and those were broken too. :(

I believe if you are trying to create a custom "block" from the onset using their new feature, that it is doable. I don't care for the block interface and am used to writing HTML from scratch (yes, very old and very old school), so I have not found their block editor useful or easy to navigate.

I have several sites I maintain and am using the Classic Widget plugin on some and the Disable Gutenberg on others. They work fine. I am tending towards liking the Disable one better as it doesn't claim to expire and it's one plugin as opposed to the two I use (Classic Editor and Classic Widgets).

To answer your question - the plugin did not fix the issue, it just prevents it from happening again. After I installed the plugin, it did away with the block, but my HTML was still broken. I was able to restore from backup and then proceeded to install the plugins on all my sites. 

A suggestion is to always have backups, turn off enabling auto backup for plugins and Core, and test updates on a staging site first. That saves a lot of grief.

I am surprised this is just coming up as WP 5.8 was released back in July.

**Reply by tech2serve** [reply]
> There appears to be a compatibility problem with widgets since WP 5.7 - I am seeing the issue reported by many people in other plugin forums. I didn't notice it until now, but the "Upcoming Meetings" function no longer works.
> 
> It will throw errors when I look at widgets. Error example: "The "tsml_widget_upcoming" block was affected by errors and may not function properly. Check the developer tools for more details."
> 
> Other widgets sometimes continue working on the sites despite errors, but in the case of TSML they are not. In my case for TSML the broken widget seems to be "Upcoming Meetings."
> 
> Sometimes the simpler ones (like "custom HTML" widgets) can be fixed by just removing and re-creating them as blocks.
> 
> Unfortunately, that is not working for this problem. Re-creating it just brings back the errors. Any help is appreciated.

Tbh, when I first saw this discussion thread, I figured @bugalug had the Gutenberg plugin installed. Not seeing that in the list of plugin, I’m scratching my head as to the cause.

I agree with Josh that it might help to see the contents of the developer console when trying to run the widget. And while it might be futile, I’d suggest disabling the other plugins and clearing the server cache to see if that resolves the issue. 

I went to what I believe is the website, and didn’t see any errors; nor did I see the upcoming meeting widget so I’m guessing it is currently disabled.

Also, please confirm for us what theme you’re using and if you’ve made any modification? Do you have a child theme loaded?

We’ll continue to try and reproduce your issue, and more information will help. Thanks for your patience.

**Reply by Grateful-Viv** [reply]
I have zero issues using the Upcoming Meetings widget while using the Classic Widget plugin.

**Reply by bugalug** [reply]
The "upcoming meetings" widget is NOT disabled, the fact that it is not working and I am unable to re-create it was the basis of my original post.

I believe the answer from Grateful-Viv sums it up best: "To answer your question - the widget did not fix the issue, it just prevents it from happening again."

I am reluctantly going to going to try the band-aid approach and use the Classic Widget plugin.

Although I could restore from backup as suggested, before I would do that I would just re-create the site from scratch, there just isn't that much to it.

**Reply by bugalug** [reply]
Oh, forgot one more thing - I meant to include the PHP checker results:

<img width="476" alt="php" src="https://user-images.githubusercontent.com/90507760/133256076-daf73feb-84c5-4f98-bff7-cbc80c93ad48.png">

**Reply by bugalug** [reply]
Classic Widget worked, looks like I will run this way until I do major rebuild, thanks everyone.
